samples/peripheral/802154_phy_test/src/ctrl/ptt_events_internal.h

```diff
@@ -0,0 +1,219 @@
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/* Purpose: events interface intended for usage inside control module */
+
+#ifndef PTT_EVENTS_INTERNAL_H__
+#define PTT_EVENTS_INTERNAL_H__
+
+#include "ctrl/ptt_events.h"
+
+#include "ptt_config.h"
+#include "ptt_errors.h"
+#include "ptt_types.h"
+#include "ptt_types_internal.h"
+#include "ptt_utils.h"
+
+/** events command type definition */
+typedef uint32_t ptt_evt_cmd_t;
+
+/** events state type definition */
+typedef uint8_t ptt_evt_state_t;
+
+PTT_COMPILE_TIME_ASSERT((PTT_EVENT_DATA_SIZE % sizeof(int32_t)) == 0u);
+
+/** struct to hold packet data and length */
+struct ptt_evt_data_s {
+	uint8_t  arr[PTT_EVENT_DATA_SIZE]; /**< packet */
+	uint16_t len;                      /**< packet used lengths */
+};
+
+PTT_COMPILE_TIME_ASSERT((PTT_EVENT_CTX_SIZE % sizeof(int32_t)) == 0u);
+
+/** struct to hold context data and length */
+struct ptt_evt_ctx_data_s {
+	uint8_t  arr[PTT_EVENT_CTX_SIZE]; /**< context */
+	uint16_t len;                     /**< context used lengths */
+};
+
+/** event data */
+struct ptt_event_s {
+	struct ptt_evt_data_s     data;  /**< packet payload */
+	struct ptt_evt_ctx_data_s ctx;   /**< command's context */
+	ptt_evt_cmd_t      cmd;   /**< command */
+	ptt_evt_state_t    state; /**< command processing state */
+	bool         use;   /**< flag to determine is this event currently in use */
```

Comment:
Again, this doesn't seem right, align all or nothing.

Reply:
Removed alignment